### PR TITLE
TD-760 Add v4 support

### PR DESCRIPTION
**Pull request by zalbiraw**

### CODE REVIEW

File: tyk-headless/templates/deployment-pmp.yaml [hunk 1]
```diff
@@ -88,37 +90,101 @@ spec:
             value: "4000"
           - name: TYK_PMP_PURGEDELAY
             value: "4"
-          {{- if .Values.mongo.enabled }}
-          - name: TYK_PMP_PUMPS_MONGO_TYPE
+{{ if .Values.backend }}
+{{ if eq .Values.backend "postgres" }}
```
Comment (zalbiraw) [inline tyk-headless/templates/deployment-pmp.yaml:94]: @vesko-tyk anyway I can simplify this? 

I tried `{{ if and .Values.backend (eq .Values.backend "postgres") }}` but it seems that the and does not short circuit and the second values causes a crash. I am trying to support older values.yamls

Comment (letzya) [reply tyk-headless/templates/deployment-pmp.yaml:94]: In go temple all parts are evaluated. You can create a function or leave it like that

File: tyk-pro/templates/deployment-dash.yaml [hunk 2]
```diff
@@ -97,13 +97,30 @@ spec:
             value: "{{ default "false" .Values.redis.useSSL }}"
           - name: TYK_DB_REDISSSLINSECURESKIPVERIFY
             value: "true"
-          - name: TYK_DB_MONGOURL
+
+{{ if and .Values.backend (eq "postgres" .Values.backend) }}
```
Comment (letzya) [inline tyk-pro/templates/deployment-dash.yaml:101]: Per comment above this needs to change to avoid crash

File: tyk-pro/values.yaml [hunk 3]
```diff
@@ -76,6 +78,21 @@ mongo:
   # Default value: false
   useSSL: false
 
+postgres:
+  # Default value: tyk-postgres-postgresql.{{ .Release.Namespace }}.svc.cluster.local
```
Comment (letzya) [inline tyk-pro/values.yaml:82]: It's a bit confusing that we write default values before the field name and also twice

**Review by letzya — DISMISSED** [review]

File: tyk-pro/values.yaml [hunk 4]
```diff
@@ -76,6 +78,21 @@ mongo:
...
+  # port: 5432
+  # Default value: postgres
+  # user: postgres
+  # password: 
+  # Default value: tyk_analytics
+  # database: tyk_analytics
+
+  # Enables SSL for MongoDB connection. MongoDB instance will have to support that.
+  # Default value: false
+  useSSL: false
```
Comment (drpebcak) [inline tyk-pro/values.yaml:94]: Is this vestigial? It references mongo in the comment, and I don't see a reference to `postgres.useSSL` anywhere. Would be nice to be able to set the sslmode for the connection string.

File: tyk-headless/templates/deployment-pmp.yaml (hunk 1 above)
Comment (zalbiraw) [reply tyk-headless/templates/deployment-pmp.yaml:94]: I see

File: tyk-pro/values.yaml (hunk 3 above)
Comment (zalbiraw) [reply tyk-pro/values.yaml:82]: Agreed

File: tyk-pro/values.yaml (hunk 4 above)
Comment (zalbiraw) [reply tyk-pro/values.yaml]: Yeah sorry copy paste error 😆

Comment (zalbiraw) [reply tyk-pro/values.yaml]: @drpebcak if I am not mistaken you are referring to the `sslmode` option on the connection string right?

Comment (drpebcak) [reply tyk-pro/values.yaml]: Yes, that's what I mean

Comment (zalbiraw) [reply tyk-pro/values.yaml]: Makes perfect sense. I am thinking I should I just allow anyone to add whatever parameters to the postgres object in the values.yaml and then collapse that to create the connection string. That should give everyone the ability to do whatever they need to do.

File: tyk-headless/values.yaml [hunk 5]
```diff
@@ -75,7 +71,16 @@ mongo:
...
+  useSSL: false
+
+# Postgres connection string parameters.
+postgres:
+  host: tyk-postgres-postgresql.tyk.svc.cluster.local
+  port: 5432
+  user: postgres
+  password:
+  database: tyk_analytics
+  sslmode:
```
Comment (zalbiraw) [inline tyk-headless/values.yaml:83]: @drpebcak now you can pass any connection string parameters in there.

**Comment by drpebcak** [comment]
After working through some other issues, this is working in my staging environment. What will it take to get this merged and released?

File: README.md [hunk 6]
```diff
@@ -4,31 +4,42 @@ Tyk provides 3 different helm charts in this repo. Please visit the respective p
...
 
 *Important Note regarding MongoDB:* This helm chart enables the PodDisruptionBudget for MongoDB with an arbiter replica-count of 1.  If you intend to perform system maintenance on the node where the MongoDB pod is running and this maintenance requires for the node to be drained, this action will be prevented due the replica count being 1.  Increase the replica count in the helm chart deployment to a minimum of 2 to remedy this issue.
 
-Another option for Redis and MongoDB, if you want to get started quickly is to use our simple charts. **Please note that these provided charts must not ever be used in production and for anything but a quick start evaluation only, use external DBs or Official Helm charts for MongoDB and Redis in any other case.**
+Another option for Redis and MongoDB or PostgreSQL, if you want to get started quickly is to use our simple charts. **Please note that these provided charts must not ever be used in production and for anything but a quick start evaluation only, use external DBs or Official Helm charts for MongoDB or PostgreSQL and Redis in any other case.**
```
Comment (komalsukhani) [inline README.md:31]: I don't think we have simple charts for postgreSQL

Comment (zalbiraw) [reply README.md:31]: No we don't

Comment (komalsukhani) [reply README.md:31]: Should we remove PostgreSQL from that part then?

**Review by komalsukhani — APPROVED** [review]

**Review by buraksekili — APPROVED** [review]

**Pull request merged**